app/services/secure_headers_allow_list.rb

```diff
@@ -1,13 +1,55 @@
 class SecureHeadersAllowList
-  def self.extract_domain(url)
-    url.split('//')[1].split('/')[0]
+  def self.csp_with_sp_redirect_uris(action_url_domain, sp_redirect_uris)
+    csp_uris = [action_url_domain]
+    csp_uris += sp_redirect_uris if sp_redirect_uris.present?
+    ["'self'"] + reduce_sp_redirect_uris_for_csp(csp_uris.compact)
```

Comment:
maybe we could just inline a few of these?
```suggestion
    ["'self'"] + reduce_sp_redirect_uris_for_csp([action_url_domain, *sp_redirect_uris])
```